src/execution/execute.js

```diff
@@ -780,9 +823,17 @@ function completeListValue(
   // where the list contains no Promises by avoiding creating another Promise.
   const itemType = returnType.ofType;
   let containsPromise = false;
-  const completedResults = result.map(item => {
-    const completedItem =
-      completeValueCatchingError(exeContext, itemType, fieldASTs, info, item);
+  const completedResults = result.map((item, index) => {
+    const childExePath = exePath.slice();
+    childExePath.push(index);
+
+    const childInfo = Object.assign({}, info, {
```

Comment:
Is this necessary? This occurs after running `resolve`, and I'm concerned about the memory overhead of copying this object on each child.